sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -61,14 +63,7 @@ class V2SessionCatalog(catalog: SessionCatalog)
   }
 
   override def loadTable(ident: Identifier): Table = {
-    val catalogTable = try {
-      catalog.getTableMetadata(ident.asTableIdentifier)
-    } catch {
-      case _: NoSuchTableException =>
```

Comment:
The caller side will try-catch the error to normalize the error class and message. We don't need to do it here.